qemu/tests/qemu_guest_agent.py

```diff
@@ -2686,6 +2686,125 @@ def bl_check(qga_cmd):
+
+        cmd_get_full_name = params["cmd_get_full_name"]
+        os_name_full_guest = session.cmd_output(cmd_get_full_name).strip()
+
+        error_context.context("Check os basic id and name.", logging.info)
+        os_type = params["os_type"]
+        os_id = params["os_id"]
+        if os_type == "windows":
+            os_name = "Microsoft Windows"
+        else:
+            os_name = re.search(r'(Red Hat.*) release',
+                                os_name_full_guest, re.I).group(1)
+        _result_check(os_id_qga, os_id)
+        _result_check(os_name_qga, os_name)
+
+        error_context.context("Check os pretty name.", logging.info)
+        if os_type == "windows":
+            os_pretty_name_guest = re.search(
+                r'Microsoft (.*)', os_name_full_guest, re.M).group(1)
+        else:
+            os_pretty_name_guest = os_name_full_guest
+            if "release" in os_name_full_guest:
+                os_pretty_name_guest = re.sub(r'release ', '',
+                                              os_name_full_guest)
+        _result_check(os_pretty_name_qga, os_pretty_name_guest)
+
+        error_context.context("Check os version info.", logging.info)
+        # 2019, 8.1, 2012 R2, 8
+        pattern = r"(\d+(.)?(?(2)(\d+))( R2)?)"
+        os_version_id_guest = re.search(pattern,
+                                        os_name_full_guest, re.I).group(1)
+        if os_type == "windows":
+            os_version_guest = re.search(r'(Microsoft.*\d)',
+                                         os_name_full_guest, re.I).group(1)
+            # 2012 R2
+            if "R2" in os_version_id_guest:
+                os_version_id_guest = re.sub(r' R2', 'R2',
+                                             os_version_id_guest)
+        else:
+            os_version_guest = re.search(r'release (\d.*)',
+                                         os_name_full_guest, re.I).group(1)
+            if "Beta" in os_version_guest:
+                os_version_guest = re.sub(r'Beta ', '', os_version_guest)
+
+        _result_check(os_version_qga, os_version_guest)
+        _result_check(os_version_id_qga, os_version_id_guest)
+
+        error_context.context("Check kernel version and release version.",
+                              logging.info)
+        cmd_get_kernel_ver = params["cmd_get_kernel_ver"]
+        kernel_info_guest = session.cmd_output(cmd_get_kernel_ver).strip()
+        if os_type == "windows":
+            kernel_g = re.search(r'(\d+\.\d+)\.(\d+)',
+                                 kernel_info_guest, re.I)
+            kernel_version_guest = kernel_g.group(1)
+            kernel_release_guest = kernel_g.group(2)
+        else:
+            kernel_version_guest = kernel_info_guest
+            cmd_get_kernel_rel = params["cmd_get_kernel_rel"]
+            kernel_release_guest = session.cmd_output(
+                cmd_get_kernel_rel).strip()
+        _result_check(kernel_version_qga, kernel_version_guest)
```

Comment:
@xiagao currently,  'uname -v' print less than information for some OS.  Are you sure it is what you need?
# uname -v
#1 SMP Tue Feb 18 16:39:12 EST 2020

Reply:
yes, it is.
{"return": {"name": "Red Hat Enterprise Linux", "kernel-release": "4.18.0-176.el8.x86_64", "version": "8.2 (Ootpa)", "pretty-name": "Red Hat Enterprise Linux 8.2 Beta (Ootpa)", "version-id": "8.2", "kernel-version": "#1 SMP Wed Feb 5 00:30:18 UTC 2020", "machine": "x86_64", "id": "rhel"}}

Reply:
Acked by: Xiangchun Fu xfu@redhat.com